internal/lock.go

```diff
@@ -105,6 +116,23 @@ func (l *Lock) DeleteResource(path string) {
 	log.Debug().Int("removed", removed).Msg("Resources deleted")
 }
 
+func deleteCache(url, token string) error {
+	req, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Authorization", "Bearer "+token)
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
```

Comment:
I think you want to move this `defer` statement up to immediately after the `client.Do()` call (so that it executes the close no matter which return path is used.